components/file_attachment_list/index.ts

```diff
@@ -6,15 +6,16 @@ import {connect} from 'react-redux';
 import {makeGetFilesForPost} from 'mattermost-redux/selectors/entities/files';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
+import {GlobalState} from 'types/store';
 import {getCurrentLocale} from 'selectors/i18n';
 import {isEmbedVisible} from 'selectors/posts';
 
-import FileAttachmentList from './file_attachment_list.jsx';
+import FileAttachmentList, {Props} from './file_attachment_list';
 
 function makeMapStateToProps() {
     const selectFilesForPost = makeGetFilesForPost();
 
-    return function mapStateToProps(state, ownProps) {
+    return function mapStateToProps(state: GlobalState, ownProps: Props) {
```

Comment:
It's generally not a good practice to use the same `Props` type that the component uses for the redux connector.
What will happen here is that any component that consumes this one will expect all props that are required for the component (eg. `locale`) to be explicitly provided by the consuming component, causing some annoying type errors.

I'd recommend here creating an `OwnProps` type specifically for what the `mapStateToProps` function needs to fetch other data and using that type instead.

Reply:
Thanks for the explanation - I've made the change you suggested.